lib/spack/spack/cmd/external.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from __future__ import print_function
+from collections import defaultdict, namedtuple
+import os
+import re
+import six
+
+import spack
+import llnl.util.tty as tty
+import spack.util.spack_yaml as syaml
+import spack.util.environment
+import llnl.util.filesystem
+
+description = "add external packages to Spack configuration"
+section = "config"
+level = "short"
+
+
+def setup_parser(subparser):
+    sp = subparser.add_subparsers(
+        metavar='SUBCOMMAND', dest='external_command')
+
+    sp.add_parser('find', help=external_find.__doc__)
```

Comment:
Just an idea, maybe for another PR, but I think in the long run we should enable users to require detection only for certain packages e.g.:
```console
$ spack external find cmake gcc
```
As more packages implement detection it might be the case that people want to register a selected subset of them.

Reply:
I like this — it would be pretty easy to do here wouldn’t it?